linetools/lists/linelist.py

```diff
@@ -593,48 +621,46 @@ def all_transitions(self, line):
         if line == 'unknown':
             return self.unknown_line()
         if self.list in ['H2']:
-
             cond = self._data['ion_name'] == line.split('(')[0]
             tbl = self._data[cond]  # cond is a masked boolean array
             if len(tbl) > 1:
                 return tbl
             else:  # this should be always len(tbl)==1 because line was found
-                return self.from_table_to_dict(tbl)
+                return lilu.from_table_to_dict(tbl)
 
         else:
             Z = None
-            for row in self._data:  # is this loop avoidable?
-                name = row['name']
-                # keep only the first part of name in linelist too
-                name = name.split(' ')[0]
-                if name == line:
-                    Z = row['Z']  # atomic number
-                    ie = row['ion']  # ionization estate
-                    Ej = row['Ej']  # Energy of lower level
-                    break
-            if Z is not None:
-                tbl = self.__getitem__((Z, ie))
-                # Make sure the lower energy level is the same too
-                #cond = tbl['Ej'] == Ej
-                cond = np.array([name1.split(' ')[0] == line for name1 in tbl['name']])
-                tbl = tbl[cond]
-                tbl.sort(['Ej','wrest'])
-                # For hydrogen/deuterium this contains deuterium/hydrogen;
-                # so let's get rid of them
-                #if (line == 'HI') or (line == 'DI'):
-                #    names = np.array(tbl['name'])
-                #    cond = np.array([l.startswith(line) for l in names])
-                #    tbl = tbl[cond]
-                if len(tbl) > 1:
-                    return tbl
-                else:  # this should be always len(tbl)==1 because Z is not None
-                    return self.from_table_to_dict(tbl)
-            else:
+            indices = np.where(self._data['ion_name'] == line)[0]
```

Comment:
See above also: self._data['ion_name'] does not exist; should be self._data['name'].  Calling, e.g., LineList.all_transitions('HI') fails currently. However, it appears that self._data['ion_name'] is set in LineList.                set_extra_columns_to_datatable()?  Either way, all_transitions() doesn't work with or without 'extras'.